### PR TITLE
CSR support in crypto HSM

### PR DESCRIPTION
**Pull request by sujaygkulkarni-nxp**
#### Problem
No CSR support in HSM layer
No HSM support for CASE protocol

#### Change overview
* Added CSR support in HSM layer
* Using P256 key from HSM for CASE operational key
* Using P256 key from HSM for CASE ephermel key
* PBKDF using HSM is disabled by default

#### Testing
* Tested manually on k32 and Linux
* Tested CSR functionality using CHIPCryptoPALTest

### CODE REVIEW

**Review by msandstedt — APPROVED** [review]

**Review by saurabhst — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
@sujaygkulkarni-nxp conflicts :(

**Comment by sujaygkulkarni-nxp** [comment]
> @sujaygkulkarni-nxp conflicts :(

Resolved.

**Comment by sujaygkulkarni-nxp** [comment]
> @sujaygkulkarni-nxp conflicts :(

Resolved the conflicts. Could you please review / merge PR.

**Comment by woody-apple** [comment]
@mspang @Damian-Nordic @LuDuda @jmartinez-silabs ?

**Comment by sujaygkulkarni-nxp** [comment]
@Reviewers -  Could you please review / merge this PR ?

File: src/crypto/hsm/nxp/CHIPCryptoPALHsm_SE05X_P256.cpp [hunk 1]
```diff
@@ -482,6 +495,216 @@ CHIP_ERROR P256PublicKeyHSM::ECDSA_validate_hash_signature(const uint8_t * hash,
...
+        memcpy((pubkey + pubKeyLen), Uint8::to_uchar(public_key), public_key.Length());
+        pubKeyLen = pubKeyLen + public_key.Length();
+    }
+
+    buffer_index = buffer_index - pubKeyLen;
+    VerifyOrExit(buffer_index > 0, error = CHIP_ERROR_INTERNAL);
+    memcpy((void *) &data_to_hash[buffer_index], pubkey, pubKeyLen);
+
+    // Copy subject (in the current implementation only organisation name info is added) and organisation OID
+    buffer_index = buffer_index - (1 + 1 + sizeof(SUBJECT_STR) - 1);
```
Comment (jmartinez-silabs) [inline src/crypto/hsm/nxp/CHIPCryptoPALHsm_SE05X_P256.cpp:593]: I don't really like how the buffer_index is decrease/calculated through this function. It seems error prone.
The use of constants could help make this clearer 
buffer_index -= kTlvHeader

**Review by jmartinez-silabs — APPROVED** [review]

File: src/crypto/hsm/nxp/CHIPCryptoPALHsm_SE05X_P256.cpp (hunk 1 above)
Comment (woody-apple) [reply src/crypto/hsm/nxp/CHIPCryptoPALHsm_SE05X_P256.cpp:593]: @sujaygkulkarni-nxp ?

Comment (sujaygkulkarni-nxp) [reply src/crypto/hsm/nxp/CHIPCryptoPALHsm_SE05X_P256.cpp]: Sure. Updated the code by using kTlvHeader,

**Comment by bzbarsky-apple** [comment]
@sujaygkulkarni-nxp is `ENABLE_HSM_CASE_EPHERMAL_KEY` supposed to be `ENABLE_HSM_CASE_EPHEMERAL_KEY`?

**Pull request merged**